### ISSUE TITLE
Classification of Cyber Bulling using NLP

### ISSUE THREAD

**Issue by abhisheks008** [World-of-ML/DL-Simplified#264]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Classification of Cyber Bulling using NLP</br>
:red_circle: **Aim** : Create a classification model based on the given dataset using NLP.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/priyanshusethi/cyber-bullying-dataset</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Abhinavcode13** [comment]
Please assign me this issue @abhisheks008

**Comment by Abhinavcode13** [comment]
Full name : Abhinav kumar
GitHub Profile Link : https://github.com/Abhinavcode13
Email ID : [abhinavkumar130503@gmail.com](mailto:abhinavkumar130503@gmail.com)
Participant ID (if applicable):
Approach for this Project :
Using NLP in the dataset

What is your participant role? (Mention the Open Source program) : SSOC 2.0

**Comment by abhisheks008** [comment]
Please inform the approach of your project. And also finish the last pull request, I have requested some changes there. After that this issue will be assigned to you.

@Abhinavcode13

**Comment by soumyadeepdas28** [comment]
Full name : Soumyadeep Das 
GitHub Profile Link : https://github.com/soumyadeepdas28
Email ID : soumyadeepdas295@gmail.com
Participant ID (if applicable):
Approach for this Project : Using NLP and Deep learning using Tensorflow
What is your participant role? (Mention the Open Source program) : SSOC 2.0

**Comment by abhisheks008** [comment]
@soumyadeepdas28 and @Nansri108 can you please confirm me the exact models you are going to use here as you have to prepare at least 3 deep learning models for this particular project.

**Comment by soumyadeepdas28** [comment]
@abhisheks008  I will use pre-trained models like **tensorflow bert** and will create an **ANN Model** and also pickup one  pre-trained model from Hugging Face

**Comment by abhisheks008** [comment]
@Nansri108 what will be your approach for solving this?

**Comment by soumyadeepdas28** [comment]
@abhisheks008 hello, Can you assign me so I  can begin ASAP?If not , please let me know

**Comment by soumyadeepdas28** [comment]
@abhisheks008  Thanks you for assigning me to the project

**Comment by soumyadeepdas28** [comment]
@abhisheks008 , Can I use another dataset for this project?

**Comment by abhisheks008** [comment]
Yeah sure. If it is providing better results than the given one, then go for it. Make sure you mention the dataset source/URL in your project.

**Comment by UtkarshTrivedi2934** [comment]
Full name : Utkarsh Trivedi
GitHub Profile Link : https://github.com/UtkarshTrivedi2934
Email ID : 21053223@kiit.ac.in
Participant ID (if applicable):
Approach for this Project : To classify cyberbullying using NLP, collecting a dataset, preprocess text data, use techniques like TF-IDF or word embeddings, choosing a suitable model, train and evaluate. Regularly monitor and update the model for ongoing effectiveness.
What is your participant role? (Mention the Open Source program) : SWOC season 4

**Comment by abhisheks008** [comment]
Issue assigned to you @UtkarshTrivedi2934

**Comment by UtkBoi69** [comment]
> Issue assigned to you @UtkarshTrivedi2934

can I use a different dataset?
[Cyberbullying_Dataset.csv](https://github.com/abhisheks008/DL-Simplified/files/13849556/Cyberbullying_Dataset.csv)

**Comment by UtkarshTrivedi2934** [comment]
> Issue assigned to you @UtkarshTrivedi2934

Hey, can I use a different dataset, like the one below?
[Uploading Cyberbullying_Dataset.csv…]()

**Comment by abhisheks008** [comment]
Which dataset you wanna use, can you share the URL?
@UtkarshTrivedi2934

**Comment by UtkarshTrivedi2934** [comment]
> Which dataset you wanna use, can you share the URL? @UtkarshTrivedi2934

Kindly find the URL below:
https://github.com/mrizwanakram/CyberBulling-With-Machine_Learning/blob/main/Cyberbullying_Dataset.csv

**Comment by abhisheks008** [comment]
It will be better if you use the Kaggle resource here, that will help others to generalize their views.
If you find the Kaggle URL of the above dataset, do share that here.

**Comment by abhisheks008** [comment]
Can you find out the dataset of your choice from Kaggle?

**Comment by UtkarshTrivedi2934** [comment]
> Can you find out the dataset of your choice from Kaggle?
Yea, I found one. Have a look
https://www.kaggle.com/code/chandrug/cyperpulling-detection/input
So basically its a JSON file, which I'll convert to CSV and then use it.

**Comment by abhisheks008** [comment]
You can go ahead with it. I have updated the dataset with the new one in the issue details.

**Comment by UtkarshTrivedi2934** [comment]
> You can go ahead with it. I have updated the dataset with the new one in the issue details.

thanks a lot, abhishek.

**Comment by aindree-2005** [comment]
Full name : Aindree Chatterjee
GitHub Profile Link : https://github.com/aindree2005
Email ID : aindree2005@gmail.com
Participant ID (if applicable):
Approach for this Project :  Trying using Bi-LSTM, BERT , and Naive Bayes Classifier
What is your participant role? (Mention the Open Source program) : `GSSOC 2024`
@abhisheks008 can I work on this?

**Comment by abhisheks008** [comment]
> Full name : Aindree Chatterjee GitHub Profile Link : https://github.com/aindree2005 Email ID : [aindree2005@gmail.com](mailto:aindree2005@gmail.com) Participant ID (if applicable): Approach for this Project : Trying using Bi-LSTM, BERT , and Naive Bayes Classifier What is your participant role? (Mention the Open Source program) : `GSSOC 2024` @abhisheks008 can I work on this?

Assigned to you @aindree-2005. Are you up with the web app too?

**Comment by aindree-2005** [comment]
> > Full name : Aindree Chatterjee GitHub Profile Link : https://github.com/aindree2005 Email ID : [aindree2005@gmail.com](mailto:aindree2005@gmail.com) Participant ID (if applicable): Approach for this Project : Trying using Bi-LSTM, BERT , and Naive Bayes Classifier What is your participant role? (Mention the Open Source program) : `GSSOC 2024` @abhisheks008 can I work on this?
> 
> Assigned to you @aindree-2005. Are you up with the web app too?

Sure I can try to make the web app, maybe once the model is done, I can do a separate PR for the app , under this issue?

**Comment by abhisheks008** [comment]
> > > Full name : Aindree Chatterjee GitHub Profile Link : https://github.com/aindree2005 Email ID : [aindree2005@gmail.com](mailto:aindree2005@gmail.com) Participant ID (if applicable): Approach for this Project : Trying using Bi-LSTM, BERT , and Naive Bayes Classifier What is your participant role? (Mention the Open Source program) : `GSSOC 2024` @abhisheks008 can I work on this?
> > 
> > 
> > Assigned to you @aindree-2005. Are you up with the web app too?
> 
> Sure I can try to make the web app, maybe once the model is done, I can do a separate PR for the app , under this issue?

Cool no issues, for the web app I'll create a new issue once you are done with this.
